milli/src/update/index_documents/extract/mod.rs

```diff
@@ -142,19 +146,21 @@ fn spawn_extraction_task<FE, FS>(
     serialize_fn: FS,
     name: &'static str,
 ) where
-    FE: Fn(grenad::Reader<CursorClonableMmap>, GrenadParameters) -> Result<grenad::Reader<File>>
+    FE: Fn(grenad::Reader<CursorClonableMmap>, GrenadParameters) -> Result<M::Output>
         + Sync
         + Send
         + 'static,
-    FS: Fn(grenad::Reader<File>) -> TypedChunk + Sync + Send + 'static,
+    FS: Fn(M::Output) -> TypedChunk + Sync + Send + 'static,
+    M: MergeableReader + FromParallelIterator<M::Output> + Send + 'static,
```

Comment:
Why do you need `FromParallelIterator`? Do we use rayon to iterate in parallel on the `Vec` here? No sure. Can you remove this constraint?

Reply:
we are doing a collect here, and according to its documentation we have:

```
pub fn collect<C>(self) -> C
where
    C: FromParallelIterator<Self::Item>,
```

what would you replace it with?